src/app/hero-detail/hero-detail.component.html

```diff
@@ -0,0 +1,10 @@
+<div *ngIf="hero">
+    <h2>{{hero.name | uppercase}} Details</h2>
+    <div><span>id: </span>{{hero.id}}</div>
+    <div>
+        <label for="hero-name">Hero name: </label>
+        <input id="hero-name" [(ngModel)]="hero.name" placeholder="Hero name"/>
+    </div>
+    <button type="button" (click)="save()">save</button>
```

Comment:
It'd be nice if this submitted a form rather than using the button click event so we could take advantage of the keyboard enter event to submit as well.